libpysal/weights/contiguity.py

```diff
@@ -183,6 +183,33 @@ def from_dataframe(cls,
                                  id_order=id_order,
                                  **kwargs)
 
+    @classmethod
+    def from_xarray(cls, da, band=None, sparse=False, **kwargs):
```

Comment:
contiguity classes are built mainly from vector based data or data containing polygons... therefore the way its constructor works is very different and it cannot build weight object from xarray_dataarray... I managed to get from_xarray working but i still doubt that its final form of the interface's API